funcx_endpoint/funcx_endpoint/endpoint/register_endpoint.py

```diff
@@ -75,3 +82,41 @@ def register_endpoint(
         )
 
     return reg_info["task_queue_info"], reg_info["result_queue_info"]
+
+
+def _get_endpoint_metadata(config: Config) -> dict:
+    metadata: t.Dict[str, t.Any] = dict()
+
+    metadata["version"] = funcx_endpoint.__version__
+
+    # let the web service get the client's IP address
+    metadata["ip_address"] = None
+    metadata["hostname"] = socket.gethostname()
```

Comment:
Given that the web service ignores `ip_address` -- summarily overwriting it -- I don't think it need be specified by the client at all.

Reply:
Just recently changed https://github.com/globusonline/funcx-services/pull/211 so that the web service no longer ignores `ip_address`. Either way, I'd argue keeping a note of that here saves future devs a brain cycle when comparing the two ends of the codebase.